### PR TITLE
Feature/about you narrative

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

**Comment by brettminnie** [comment]
Am aware of the constants in the AboutYouNarrativeBuilder, need to figure out a clean solution for the substitution of strings without making the builder container aware

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Section/AboutYou/Feeling.php [hunk 1]
```diff
@@ -189,18 +189,33 @@ public function setMood($mood)
...
     public function getNormalisedValues()
     {
         return array(
-            'energy'  => (int)($this->energy / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'fatigue' => (int)($this->fatigue / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'fitness' => (int)($this->fitness / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'mood'    => (int)($this->mood / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'stress'  => (int)($this->stress / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'weight'  => (int)($this->weight / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
+            'energy'  => (int)(($this->energy-1) / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Section/AboutYou/Feeling.php]: why `-1`?

Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Section/AboutYou/Feeling.php]: Read the docblock :)

Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Section/AboutYou/Feeling.php]: Use a `const` not a magic number ;)

Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Section/AboutYou/Feeling.php]: there is nothing magic about -1 :D

Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Section/AboutYou/Feeling.php]: Your call, but `-1` is quite magic, const would be better, similar to the other ones ;)

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/NarrativeBuilderAbstract.php [hunk 2]
```diff
@@ -10,11 +10,11 @@
...
-    const CONST_SYMPTOM_MAX  = 2;
-    const CONST_SYMPTOM_HIGH = 1;
-    const CONST_SYMPTOM_AVG  = 0;
-    const CONST_SYMPTOM_LOW  = -1;
-    const CONST_SYMPTOM_MIN  = -2;
+    const SYMPTOM_MAX  = 2;
+    const SYMPTOM_HIGH = 1;
+    const SYMPTOM_AVG  = 0;
+    const SYMPTOM_LOW  = -1;
+    const SYMPTOM_MIN  = -2;
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/NarrativeBuilderAbstract.php]: :+1:

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Section/AboutYou/Feeling.php [hunk 3]
```diff
@@ -189,18 +191,39 @@ public function setMood($mood)
...
     public function getNormalisedValues()
     {
         return array(
-            'energy'  => (int)($this->energy / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'fatigue' => (int)($this->fatigue / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'fitness' => (int)($this->fitness / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'mood'    => (int)($this->mood / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'stress'  => (int)($this->stress / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
-            'weight'  => (int)($this->weight / self::NORMALIZATION_DIVISOR) - self::NORMALIZATION_SHIFT,
+            'energy'  => (int)(($this->energy - self::NORMALIZATION_OFFSET)
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Questions/Section/AboutYou/Feeling.php]: YES :+1:

**Pull request merged**